app/controllers/hosts_controller.rb

```diff
@@ -274,7 +274,16 @@ def overview
   end
 
   def bmc
-    render :partial => 'bmc', :locals => { :host => @host }
+    render :partial => 'bmc', :locals => {
+      status: power_status(@host.power.state),
+      bmc_proxy: @host.bmc_proxy,
+    }
+  rescue Foreman::Exception => exception
+    if exception.message =~ /ERF42-9474/
```

Comment:
Thinking out loud: can we store this code as a property of the exception rather than doing a regex on the message?

Reply:
Really good point, the correct approach is to define our own exception. Will do that: `Foreman::BMCFeatureException`.

Reply:
I like that. Thinking about it, perhaps it needs a parent class `ProxyFeatureException` since I think it can be a common pattern that something needs a feature, but it isn't available.